edge-util/src/Microsoft.Azure.Devices.Edge.Storage/StoreProvider.cs

```diff
@@ -25,10 +25,22 @@ public StoreProvider(IDbStoreProvider dbStoreProvider, TimeSpan operationTimeout
         public IEntityStore<TK, TV> GetEntityStore<TK, TV>(string entityName)
         {
             IDbStore entityDbStore = this.dbStoreProvider.GetDbStore(Preconditions.CheckNonWhiteSpace(entityName, nameof(entityName)));
-            IKeyValueStore<TK, TV> dbStoreMapper = new KeyValueStoreMapper<TK, byte[], TV, byte[]>(entityDbStore, new BytesMapper<TK>(), new BytesMapper<TV>());
-            IEntityStore<TK, TV> entityStore = new EntityStore<TK, TV>(dbStoreMapper, entityName, 12);
-            IEntityStore<TK, TV> timedEntityStore = new TimedEntityStore<TK, TV>(entityStore, this.operationTimeout);
-            return timedEntityStore;
+            return this.GetEntityStore<TK, TV>(entityName, entityDbStore);
+        }
+
+        public IEntityStore<TK, TV> GetEntityStore<TK, TV>(string backwardCompatibleEntityName, string entityName)
```

Comment:
I wonder if we should make this logic specific to the TwinStore..

Reply:
I think the store provider should handle returning the store so it looks like more of a provider logic.  On the other way it's true that this is specific for twin because we need to get around having 2 names.